src/Mutator/Boolean/LogicalOr.php

```diff
@@ -77,6 +79,72 @@ public function mutate(Node $node): iterable
 
     public function canMutate(Node $node): bool
     {
-        return $node instanceof Node\Expr\BinaryOp\BooleanOr;
+        if (!$node instanceof Node\Expr\BinaryOp\BooleanOr) {
+            return false;
+        }
+
+        if (
+            (
+                $node->left instanceof Node\Expr\BinaryOp\Identical
+                && $node->right instanceof Node\Expr\BinaryOp\Identical
+            ) || (
+                $node->left instanceof Node\Expr\BinaryOp\Equal
+                && $node->right instanceof Node\Expr\BinaryOp\Equal
+            )
+        ) {
+            $varName = '';
+
+            if ($node->left->left instanceof Node\Expr\Variable) {
+                $varName = $node->left->left->name;
+            } elseif ($node->left->right instanceof Node\Expr\Variable) {
+                $varName = $node->left->right->name;
+            }
+
+            if ($node->right->left instanceof Node\Expr\Variable) {
+                return $varName !== $node->right->left->name;
+            } elseif ($node->right->right instanceof Node\Expr\Variable) {
+                return $varName !== $node->right->right->name;
+            }
+        }
+
+        $greaterOp = [
+            Node\Expr\BinaryOp\Greater::class,
+            Node\Expr\BinaryOp\GreaterOrEqual::class,
+        ];
+
+        $smallerOp = [
+            Node\Expr\BinaryOp\Smaller::class,
+            Node\Expr\BinaryOp\SmallerOrEqual::class,
+        ];
+
+        if (
+            (
+                in_array(get_class($node->left), $greaterOp, true) === true
```

Comment:
Could have applied this logic only once but I figured if we need to compare the values too then the logic might differ